merino-web/src/extractors.rs

```diff
@@ -86,85 +84,29 @@ struct SuggestQuery {
 struct SupportedLanguagesWrapper(SupportedLanguages);
 
 impl FromRequest for SupportedLanguagesWrapper {
-    type Config = ();
     type Error = ActixError;
     type Future = Ready<Result<Self, Self::Error>>;
 
     fn from_request(req: &HttpRequest, _payload: &mut Payload) -> Self::Future {
-        /// Parse the quality value from a string of the form q=`<quality value>`.
-        fn parse_quality_value(quality_value: &str) -> Result<f64, HandlerError> {
-            let (_, weight_as_string) = quality_value
-                .split_once('=')
-                .ok_or(HandlerErrorKind::MalformedHeader("Accept-Language"))?;
-
-            let weight = weight_as_string
-                .parse::<f64>()
-                .map_err(|_| HandlerErrorKind::MalformedHeader("Accept-Language"))?;
-
-            if (0.0..=1.0).contains(&weight) {
-                Ok(weight)
-            } else {
-                Err(HandlerErrorKind::MalformedHeader("Accept-Language").into())
-            }
-        }
-
-        /// Parse the Accept-Language HTTP header.
-        fn parse_language(raw_language: &str) -> Result<Language, ActixError> {
-            let (locale_or_wildcard, quality_value) =
-                if let Some((language, quality_value)) = raw_language.split_once(';') {
-                    let quality_value = Some(parse_quality_value(quality_value)?);
-
-                    (language, quality_value)
-                } else {
-                    (raw_language, None)
-                };
-
-            let language = if locale_or_wildcard == "*" {
-                Language {
-                    language_identifier: LanguageIdentifier::Wildcard,
-                    quality_value,
-                }
-            } else if let Some((language, region)) = locale_or_wildcard.split_once("-") {
-                Language {
-                    language_identifier: LanguageIdentifier::Locale {
-                        language: language.to_lowercase(),
-                        region: Some(region.to_lowercase()),
-                    },
-                    quality_value,
+        future::ready({
+            if req.headers().contains_key("Accept-Language") {
+                match AcceptLanguage::parse(req) {
```

Comment:
`AcceptLanguage::parse` just returns `AcceptLanguage([])` for most types of invalid headers, but it returns the same thing when the request doesn't have an `Accept-Language` header. I handle that by explicitly checking if the header is present and handling the case separately.